### ISSUE TITLE
Filter to allow push/pull of draft/scheduled posts in network/external connected sites

### ISSUE THREAD

**Issue by garman22** [10up/distributor#1149]
[Edited 2023-11-29 by @jeffpaul]
I'm updating the title and the description in this area here to note we should review the push/pull options to allow for a filter that is disabled/false by default but could be set to true that would enable drafts and scheduled posts to be able to be pushed/pulled in a network and external connected sites.

### Describe the bug

In previous versions (for example 1.9.1 for sure) we were able to view both published and draft posts in the Pull Content list on the recipient website.   In version 2.0.1 we can no longer see any posts that are 'draft' status in the Pull Content list.  Posts only become available to pull once they are published.

Is this feature change intended?  Or a bug.
It's an issue for us, since we sometimes wish to pull posts from one source to another before they are published.
If there's a workaround (or a hook) that would be nice. 

### Steps to Reproduce

On the receiving site go to Distributor > Pull Content > Pull Content from (choose your site) > New.
We see only posts that have either been published on the source website or pulled posts.  No posts in draft status are available to pull. 

### WordPress information

WordPress 6.3.2
Distributor 2.0.1 (on all sites)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by peterwilsoncc** [comment]
There have been some reports of this affecting scheduled posts too, ie other unpublished post statuses, so I'll investigate the issue for both draft and future statuses. It's probable that it affects other statuses such as pending and custom statuses.

**Comment by peterwilsoncc** [comment]
@garman22 I've looked in to your issue.

In Distributor 1.9.1 there was a difference on the pull screen for different types of connections:

* Internal (Multisite) connections would show the draft posts
* External (via the REST API) connections would not show the draft posts

This inconsistency has remained in Distributor 2.0 and we did find some bugs with pulling unpublished posts on multisite connections.

Is it possible you were seeing the draft posts on an internal connection previously and noticed this issue you are seeing on an external connection? 

If you were using a plugin to display drafts via the REST API, are you able to share the plugin with us so we can investigate further?
